### ISSUE TITLE
K2: unresolved reference errors when generated code references project symbols

### ISSUE THREAD

**Issue by rickclephas** [google/ksp#1570]
Related to #1569, with KSP `1.9.20-RC-1.0.13` compilations using K2 fail when the generated code references project symbols.

```kotlin
// project code
class MyClass {
   val myProp: String = "my property"
}

// KSP generated code
val MyClass.kspProp: String get() = "$myProp by KSP"
```

Building with K2 enabled:
```properties
kotlin.experimental.tryK2=true
```

results in:
```
e: Unresolved reference: MyClass
e: Unresolved reference: myProp
```

Also see [KT-60881](https://youtrack.jetbrains.com/issue/KT-60881).

**Comment by ting-yuan** [comment]
Can you share how sources are organized in the project? Specifically, are they in the same source set? K2 forbids the references from common source sets to target source sets and it is one of the possible reasons for this error.

**Comment by rickclephas** [comment]
@ting-yuan in the above example `MyClass` is in `commonMain` and the generated `kspProp` is in the KSP target source set (e.g. `generatedByKspKotlinIosSimulatorArm64Main`).

> K2 forbids the references from common source sets to target source sets and it is one of the possible reasons for this error.

I am not exactly sure what you mean by that. If I add the `kspProp` to the projects `iosSimulatorArm64Main` source set (without using KSP) the project compiles just fine.

It seems the KSP source sets are actually missing a source set dependency. The project does compile once I add this:
```kotlin
val iosMain by getting 
afterEvaluate {
    val generatedByKspKotlinIosSimulatorArm64 by getting {
        dependsOn(iosMain)
    }
}
```
That doesn't really solve it though since you [can't use](https://youtrack.jetbrains.com/issue/KT-60881/Errors-building-KSP-generated-code-in-PeopleInSpace-sample-when-K2-compiler-enabled#focus=Comments-27-8243481.0-0) the target's source set (only intermediates).

Anyway I think it's related to #1569, so this might be fixed by #1575 (haven't tested it though).

**Comment by rickclephas** [comment]
Just tested with `1.9.20-RC2-1.0.13` and I can confirm this has been fixed (by #1575).
